### DISCUSSION TITLE
Label in Provider

### DISCUSSION THREAD

**Discussion by andybuki** [ksclarke/jiiify-presentation#131] [Q&A] [answered]
Hi Kevin. 
I have a new question:

I try to add some providers to my manifest with the following structure:

```
"provider": [
    {
      "id": "crossasia.org",
      "type": "Agent",
      "label": {
        "none": [
          "Staatsbibliothek zu Berlin | CrossAsia"
        ]
      }
    },
    {
      "id": "http://nationallibraryoflaos.net/en/",
      "type": "Agent",
      "label": {
        "en": [
          "National Library of Laos"
        ],
        "lo": [
          "ຫໍສະໝຸດແຫ່ງຊາດ"
        ]
      }
    }
  ],
```

In this case it's 2 providers. It is easy to add the first one, but if i try to add a second one with different languages then i need to wrap it to string.

I try to right something like:
```
Provider provider = new Provider("http://nationallibraryoflaos.net/en/",
                   (new Label(new I18n("en", "National Library of Laos"),new I18n("lo", "ຫໍສະໝຸດແຫ່ງຊາດ"))));
```

Do you have some ideas how to realise it.

Best regards
Andrey

**Comment by andybuki** [comment, marked as answer]
Hi Kevin. 
I found how to do it.

Best regards
Andrey

**Reply by ksclarke** [reply]
Glad you already figured it out. Just for others who might stumble on this discussion forum, I'll add one way to do it here for documentation purposes:
```
Provider provider = new Provider(URI.create("http://nationallibraryoflaos.net/en/"),
                new Label("en", "National Library of Laos").addI18ns(new I18n("lo", "ຫໍສະໝຸດແຫ່ງຊາດ")));
```
As an aside, I do see it would be good to add an additional constructor so one doesn't _have_ to create the URI themselves... so something like the following is possible:
```
Provider provider = new Provider("http://nationallibraryoflaos.net/en/",
                new Label("en", "National Library of Laos").addI18ns(new I18n("lo", "ຫໍສະໝຸດແຫ່ງຊາດ")));
```
I'll create a ticket to add that: #132 

Thanks for asking the question.

**Reply by ksclarke** [reply]
... And another way (that's closer to your original example and that, perhaps, doesn't seem to give preference to any one internationalization (i.e. doesn't make it look like there is a "default" language)):
```
Provider provider = new Provider(URI.create("http://nationallibraryoflaos.net/en/"),
                new Label(new I18n("en", "National Library of Laos"), new I18n("lo", "ຫໍສະໝຸດແຫ່ງຊາດ")));
```
:-)
